#! /bin/sh /usr/share/dpatch/dpatch-run
## 002-nant-use-system-dlls.dpatch by Jelmer Vernooij <jelmer@samba.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix build files to use system-provided libraries rather than 
## DP: the libraries shipped inside nant's tarball.

@DPATCH@
diff -ur nant-0.85/Makefile nant-0.85dfsg1/Makefile
--- nant-0.85/Makefile	2005-12-11 09:47:54.000000000 +0100
+++ nant-0.85dfsg1/Makefile	2008-05-26 23:48:23.000000000 +0200
@@ -46,7 +46,7 @@
 	$(NANT) -f:NAnt.build test
 	
 bootstrap/NAnt.exe:
-	$(MCS) -target:exe -define:${DEFINE} -out:bootstrap${DIRSEP}NAnt.exe -r:bootstrap${DIRSEP}log4net.dll \
+	$(MCS) -target:exe -define:${DEFINE} -out:bootstrap${DIRSEP}NAnt.exe -pkg:log4net \
 		-recurse:src${DIRSEP}NAnt.Console${DIRSEP}*.cs src${DIRSEP}CommonAssemblyInfo.cs
 	
 
@@ -55,28 +55,26 @@
 
 setup:
 	mkdir -p bootstrap
-	cp -R lib/ bootstrap/lib
 	# Mono loads log4net before privatebinpath is set-up, so we need this in the same directory
 	# as NAnt.exe
-	cp lib/log4net.dll bootstrap
 	cp src/NAnt.Console/App.config bootstrap/NAnt.exe.config
 
 bootstrap/NAnt.Core.dll:
 	$(RESGEN)  src/NAnt.Core/Resources/Strings.resx bootstrap/NAnt.Core.Resources.Strings.resources
 	$(MCS) -target:library -warn:0 -define:${DEFINE} -out:bootstrap/NAnt.Core.dll -debug \
-		-resource:bootstrap/NAnt.Core.Resources.Strings.resources -r:lib${DIRSEP}log4net.dll \
+		-resource:bootstrap/NAnt.Core.Resources.Strings.resources -pkg:log4net \
 		-r:System.Web.dll -recurse:src${DIRSEP}NAnt.Core${DIRSEP}*.cs src${DIRSEP}CommonAssemblyInfo.cs
 
 bootstrap/NAnt.DotNetTasks.dll:
 	$(RESGEN)  src/NAnt.DotNet/Resources/Strings.resx bootstrap/NAnt.DotNet.Resources.Strings.resources
 	$(MCS) -target:library -warn:0 -define:MONO -out:bootstrap/NAnt.DotNetTasks.dll \
-		-r:./bootstrap/NAnt.Core.dll -r:bootstrap/lib/${FRAMEWORK_DIR}/1.0/NDoc.Core.dll \
+		-r:./bootstrap/NAnt.Core.dll -r:/usr/lib/ndoc-1.3/NDoc.Core.dll \
 		-recurse:src${DIRSEP}NAnt.DotNet${DIRSEP}*.cs -resource:bootstrap/NAnt.DotNet.Resources.Strings.resources \
 		src${DIRSEP}CommonAssemblyInfo.cs
 
 bootstrap/NAnt.CompressionTasks.dll:
 	$(MCS) -target:library -warn:0 -define:MONO -out:bootstrap/NAnt.CompressionTasks.dll \
-		-r:./bootstrap/NAnt.Core.dll -r:bootstrap/lib/ICSharpCode.SharpZipLib.dll \
+		-r:./bootstrap/NAnt.Core.dll -r:ICSharpCode.SharpZipLib.dll \
 		-recurse:src${DIRSEP}NAnt.Compression${DIRSEP}*.cs src${DIRSEP}CommonAssemblyInfo.cs
 
 bootstrap/NAnt.Win32Tasks.dll:
diff -ur nant-0.85/NAnt.build nant-0.85dfsg1/NAnt.build
--- nant-0.85/NAnt.build	2006-10-14 16:01:34.000000000 +0200
+++ nant-0.85dfsg1/NAnt.build	2008-05-27 00:08:48.000000000 +0200
@@ -38,7 +38,7 @@
         <include name="NAnt.DotNet/NAnt.DotNet.build" />
         <include name="NAnt.Compression/NAnt.Compression.build" />
         <include name="NAnt.NUnit/NAnt.NUnit.build" />
-        <include name="NAnt.SourceControl/NAnt.SourceControl.build" />
+		<!--<include name="NAnt.SourceControl/NAnt.SourceControl.build" />-->
         <include name="NAnt.MSNet/NAnt.MSNet.build" />
         <include name="NAnt.Win32/NAnt.Win32.build" />
         <include name="NAnt.VisualCpp/NAnt.VisualCpp.build" />
@@ -48,7 +48,7 @@
     <fileset id="ext.core.tests" basedir="tests">
         <include name="NAnt.DotNet/NAnt.DotNet.build" />
         <include name="NAnt.Compression/NAnt.Compression.build" />
-        <include name="NAnt.SourceControl/NAnt.SourceControl.build" />
+		<!--<include name="NAnt.SourceControl/NAnt.SourceControl.build" />-->
         <include name="NAnt.MSNet/NAnt.MSNet.build" />
         <include name="NAnt.Win32/NAnt.Win32.build" />
         <include name="NAnt.VisualCpp/NAnt.VisualCpp.build" />
diff -ur nant-0.85/src/NAnt.Compression/NAnt.Compression.build nant-0.85dfsg1/src/NAnt.Compression/NAnt.Compression.build
--- nant-0.85/src/NAnt.Compression/NAnt.Compression.build	2006-03-25 13:11:12.000000000 +0100
+++ nant-0.85dfsg1/src/NAnt.Compression/NAnt.Compression.build	2008-05-26 23:59:07.000000000 +0200
@@ -19,7 +19,7 @@
                 <include name="${build.dir}/bin/log4net.dll" />
                 <!-- end workaround -->
                 <include name="${build.dir}/bin/NAnt.Core.dll" />
-                <include name="${lib.dir}/ICSharpCode.SharpZipLib.dll" />
+                <include name="ICSharpCode.SharpZipLib.dll" />
             </references>
             <resources>
                 <include name="*.resx"/>
diff -ur nant-0.85/src/NAnt.Console/NAnt.Console.build nant-0.85dfsg1/src/NAnt.Console/NAnt.Console.build
--- nant-0.85/src/NAnt.Console/NAnt.Console.build	2005-02-20 13:29:50.000000000 +0100
+++ nant-0.85dfsg1/src/NAnt.Console/NAnt.Console.build	2008-05-26 23:50:33.000000000 +0200
@@ -27,9 +27,11 @@
                 <include name="../CommonAssemblyInfo.cs" />
             </sources>
             <references>
-                <include name="${build.dir}/bin/log4net.dll" />
                 <include name="System.Xml.dll" />
             </references>
+			<pkg-references>
+				<package name="log4net"/>
+			</pkg-references>
             <resources>
                 <include name="*.resx"/>
             </resources>
diff -ur nant-0.85/src/NAnt.Core/NAnt.Core.build nant-0.85dfsg1/src/NAnt.Core/NAnt.Core.build
--- nant-0.85/src/NAnt.Core/NAnt.Core.build	2006-03-18 07:37:34.000000000 +0100
+++ nant-0.85dfsg1/src/NAnt.Core/NAnt.Core.build	2008-05-26 23:48:23.000000000 +0200
@@ -24,9 +24,11 @@
                 <include name="Resources/**/*" />
             </resources>
             <references>
-                <include name="${build.dir}/bin/log4net.dll"/>
                 <include name="System.Web.dll"/>
             </references>
+			<pkg-references>
+				<package name="log4net"/>
+			</pkg-references>
         </csc>
     </target>
 </project>
diff -ur nant-0.85/src/NAnt.DotNet/NAnt.DotNet.build nant-0.85dfsg1/src/NAnt.DotNet/NAnt.DotNet.build
--- nant-0.85/src/NAnt.DotNet/NAnt.DotNet.build	2006-03-18 07:37:34.000000000 +0100
+++ nant-0.85dfsg1/src/NAnt.DotNet/NAnt.DotNet.build	2008-05-26 23:56:25.000000000 +0200
@@ -20,8 +20,8 @@
             </sources>
             <references>
                 <include name="${build.dir}/bin/NAnt.Core.dll" />
-                <include name="${lib.framework.dir}/NDoc.ExtendedUI.dll" />
-                <include name="${lib.framework.dir}/NDoc.Core.dll" />
+                <include name="/usr/lib/ndoc-1.3/NDoc.ExtendedUI.dll" />
+				<include name="/usr/lib/ndoc-1.3/NDoc.Core.dll" />
             </references>
             <resources prefix="NAnt.DotNet" dynamicprefix="true">
                 <include name="Resources/**/*" />
diff -ur nant-0.85/src/NAnt.NUnit/NAnt.NUnit.build nant-0.85dfsg1/src/NAnt.NUnit/NAnt.NUnit.build
--- nant-0.85/src/NAnt.NUnit/NAnt.NUnit.build	2005-02-20 13:29:50.000000000 +0100
+++ nant-0.85dfsg1/src/NAnt.NUnit/NAnt.NUnit.build	2008-05-27 00:04:20.000000000 +0200
@@ -16,32 +16,10 @@
             </sources>
             <references>
                 <include name="${build.dir}/bin/NAnt.Core.dll" />
-                <include name="${lib.framework.dir}/nunit.framework.dll" />
             </references>
-        </csc>
-        <csc target="library" define="${current.build.defines}" warnaserror="true" debug="${build.debug}" output="${build.dir}/bin/NAnt.NUnit1Tasks.dll" doc="${build.dir}/bin/NAnt.NUnit1Tasks.xml">
-            <nowarn>
-                <!-- do not report warnings for missing XML comments -->
-                <warning number="1591" />
-                 <!-- workaround for Mono bug #61902: do not report deprecation warnings -->
-                <warning number="0618" if="${framework::get-family(framework::get-target-framework()) == 'mono'}" />
-            </nowarn>
-            <sources>
-                <include name="NUnit1/**/*.cs"/>
-                <!-- common assembly-level attributes -->
-                <include name="../CommonAssemblyInfo.cs" />
-            </sources>
-            <references>
-                <!-- temporary workaround for Mono Runtime bug #57602 -->
-                <include name="${lib.framework.dir}/log4net.dll" />
-                <!-- end workaround -->
-                <include name="${build.dir}/bin/NAnt.Core.dll" />
-                <include name="${build.dir}/bin/NAnt.NUnit.dll" />
-                <include name="${lib.dir}/NUnitCore.dll" />
-            </references>
-            <resources>
-                <include name="*.resx"/>
-            </resources>
+			<pkg-references>
+				<package name="nunit"/>
+			</pkg-references>
         </csc>
         <csc target="library" define="${current.build.defines}" warnaserror="true" debug="${build.debug}" output="${build.dir}/bin/NAnt.NUnit2Tasks.dll" doc="${build.dir}/bin/NAnt.NUnit2Tasks.xml">
             <nowarn>
@@ -55,16 +33,16 @@
             </sources>
             <references>
                 <!-- temporary workaround for Mono runtime bug #57602 -->
-                <include name="${lib.framework.dir}/log4net.dll" />
-                <include name="${lib.framework.dir}/NDoc.Core.dll" />
+                <include name="NDoc.Core.dll" />
                 <!-- end workaround -->
                 <include name="${build.dir}/bin/NAnt.Core.dll" />
                 <include name="${build.dir}/bin/NAnt.NUnit.dll" />
                 <include name="${build.dir}/bin/NAnt.DotNetTasks.dll" />
-                <include name="${lib.framework.dir}/nunit.framework.dll" />
-                <include name="${lib.framework.dir}/nunit.core.dll" />
-                <include name="${lib.framework.dir}/nunit.util.dll" />
             </references>
+			<pkg-references>
+				<package name="nunit"/>
+				<package name="log4net"/>
+			</pkg-references>
         </csc>
     </target>
 </project>
diff -ur nant-0.85/src/NAnt.SourceControl/NAnt.SourceControl.build nant-0.85dfsg1/src/NAnt.SourceControl/NAnt.SourceControl.build
--- nant-0.85/src/NAnt.SourceControl/NAnt.SourceControl.build	2006-03-18 07:37:34.000000000 +0100
+++ nant-0.85dfsg1/src/NAnt.SourceControl/NAnt.SourceControl.build	2008-05-27 00:05:32.000000000 +0200
@@ -15,11 +15,13 @@
                 <include name="../CommonAssemblyInfo.cs" />
             </sources>
             <references>
-                <include name="${build.dir}/bin/log4net.dll" />
                 <include name="${build.dir}/bin/NAnt.Core.dll" />
-                <include name="${lib.dir}/ICSharpCode.SharpZipLib.dll" />
-                <include name="${lib.dir}/ICSharpCode.SharpCvsLib.dll" />
+                <include name="ICSharpCode.SharpZipLib.dll" />
+                <include name="ICSharpCode.SharpCvsLib.dll" />
             </references>
+			<pkg-references>
+				<package name="log4net"/>
+			</pkg-references>
             <resources>
                 <include name="*.resx"/>
             </resources>
diff -ur nant-0.85/tests/NAnt.Compression/NAnt.Compression.build nant-0.85dfsg1/tests/NAnt.Compression/NAnt.Compression.build
--- nant-0.85/tests/NAnt.Compression/NAnt.Compression.build	2006-03-18 07:37:34.000000000 +0100
+++ nant-0.85dfsg1/tests/NAnt.Compression/NAnt.Compression.build	2008-05-27 00:11:22.000000000 +0200
@@ -20,12 +20,14 @@
             </sources>
             <references>
                 <!-- temporary workaround for Mono Runtime bug #57602 -->
-                <include name="${lib.framework.dir}/log4net.dll" />
                 <!-- end workaround -->
                 <include name="${build.dir}/bin/NAnt.Core.dll" />
                 <include name="${build.dir}/bin/NAnt.Core.Tests.dll" />
-                <include name="${lib.framework.dir}/nunit.framework.dll" />
             </references>
+			<pkg-references>
+				<package name="log4net"/>
+				<package name="nunit"/>
+			</pkg-references>
         </csc>
     </target>
     <target name="test" depends="build">
diff -ur nant-0.85/tests/NAnt.Console/NAnt.Console.build nant-0.85dfsg1/tests/NAnt.Console/NAnt.Console.build
--- nant-0.85/tests/NAnt.Console/NAnt.Console.build	2006-03-18 07:37:34.000000000 +0100
+++ nant-0.85dfsg1/tests/NAnt.Console/NAnt.Console.build	2008-05-27 00:11:31.000000000 +0200
@@ -20,11 +20,13 @@
                 <include name="../../src/CommonAssemblyInfo.cs" />
             </sources>
             <references>
-                <include name="${lib.framework.dir}/log4net.dll" />
                 <include name="${build.dir}/bin/NAnt.Core.dll" />
                 <include name="${build.dir}/bin/NAnt.Core.Tests.dll" />
-                <include name="${lib.framework.dir}/nunit.framework.dll" />
             </references>
+			<pkg-references>
+				<package name="log4net"/>
+				<package name="nunit"/>
+			</pkg-references>
         </csc>
     </target>
     <target name="test" depends="build">
diff -ur nant-0.85/tests/NAnt.Core/NAnt.Core.build nant-0.85dfsg1/tests/NAnt.Core/NAnt.Core.build
--- nant-0.85/tests/NAnt.Core/NAnt.Core.build	2006-03-18 07:37:34.000000000 +0100
+++ nant-0.85dfsg1/tests/NAnt.Core/NAnt.Core.build	2008-05-27 00:10:47.000000000 +0200
@@ -21,10 +21,12 @@
                 <include name="../../src/CommonAssemblyInfo.cs" />
             </sources>
             <references>
-                <include name="${lib.framework.dir}/log4net.dll" />
                 <include name="${build.dir}/bin/NAnt.Core.dll" />
-                <include name="${lib.framework.dir}/nunit.framework.dll" />
             </references>
+			<pkg-references>
+				<package name="nunit"/>
+				<package name="log4net"/>
+			</pkg-references>
             <resources failonempty="true" basedir="Resources" dynamicprefix="true" prefix="XML:">
                 <include name="**/*.xml"/>
                 <include name="**/*.build"/>
diff -ur nant-0.85/tests/NAnt.DotNet/NAnt.DotNet.build nant-0.85dfsg1/tests/NAnt.DotNet/NAnt.DotNet.build
--- nant-0.85/tests/NAnt.DotNet/NAnt.DotNet.build	2006-05-31 20:16:26.000000000 +0200
+++ nant-0.85dfsg1/tests/NAnt.DotNet/NAnt.DotNet.build	2008-05-27 00:12:01.000000000 +0200
@@ -27,14 +27,16 @@
             </sources>
             <references>
                 <!-- temporary workaround for Mono Runtime bug #57602 -->
-                <include name="${lib.framework.dir}/log4net.dll" />
-                <include name="${lib.framework.dir}/NDoc.Core.dll" />
+				<include name="/usr/lib/ndoc-1.3/NDoc.Core.dll" />
                 <!-- end workaround -->
                 <include name="${build.dir}/bin/NAnt.Core.dll"/>
                 <include name="${build.dir}/bin/NAnt.Core.Tests.dll"/>
                 <include name="${build.dir}/bin/${project::get-name()}Tasks.dll"/>
-                <include name="${lib.framework.dir}/nunit.framework.dll"/>
             </references>
+			<pkg-references>
+				<package name="log4net"/>
+				<package name="nunit"/>
+			</pkg-references>
         </csc>
     </target>
     <target name="test" depends="build">
diff -ur nant-0.85/tests/NAnt.MSNet/NAnt.MSNet.build nant-0.85dfsg1/tests/NAnt.MSNet/NAnt.MSNet.build
--- nant-0.85/tests/NAnt.MSNet/NAnt.MSNet.build	2006-09-03 08:56:22.000000000 +0200
+++ nant-0.85dfsg1/tests/NAnt.MSNet/NAnt.MSNet.build	2008-05-27 00:12:17.000000000 +0200
@@ -21,8 +21,10 @@
             <references>
                 <include name="${build.dir}/bin/NAnt.Core.dll" />
                 <include name="${build.dir}/bin/NAnt.Core.Tests.dll" />
-                <include name="${lib.framework.dir}/nunit.framework.dll" />
             </references>
+			<pkg-references>
+				<package name="nunit"/>
+			</pkg-references>
         </csc>
     </target>
     <target name="test" depends="build" if="${framework::get-family(framework::get-target-framework()) == 'net'}">
diff -ur nant-0.85/tests/NAnt.VisualCpp/NAnt.VisualCpp.build nant-0.85dfsg1/tests/NAnt.VisualCpp/NAnt.VisualCpp.build
--- nant-0.85/tests/NAnt.VisualCpp/NAnt.VisualCpp.build	2006-09-03 08:57:06.000000000 +0200
+++ nant-0.85dfsg1/tests/NAnt.VisualCpp/NAnt.VisualCpp.build	2008-05-27 00:14:39.000000000 +0200
@@ -22,8 +22,10 @@
                 <include name="${build.dir}/bin/NAnt.Core.dll" />
                 <include name="${build.dir}/bin/NAnt.Core.Tests.dll" />
                 <include name="${build.dir}/bin/NAnt.VisualCppTasks.dll" />
-                <include name="${lib.framework.dir}/nunit.framework.dll" />
             </references>
+			<pkg-references>
+				<package name="nunit"/>
+			</pkg-references>
         </csc>
     </target>
     <target name="test" depends="build" if="${platform::is-win32()}">
diff -ur nant-0.85/tests/NAnt.VSNet/NAnt.VSNet.build nant-0.85dfsg1/tests/NAnt.VSNet/NAnt.VSNet.build
--- nant-0.85/tests/NAnt.VSNet/NAnt.VSNet.build	2006-09-03 08:57:52.000000000 +0200
+++ nant-0.85dfsg1/tests/NAnt.VSNet/NAnt.VSNet.build	2008-05-27 00:16:01.000000000 +0200
@@ -22,8 +22,10 @@
                 <include name="${build.dir}/bin/NAnt.Core.dll" />
                 <include name="${build.dir}/bin/NAnt.Core.Tests.dll" />
                 <include name="${build.dir}/bin/NAnt.VisualCpp.Tests.dll" />
-                <include name="${lib.framework.dir}/nunit.framework.dll" />
             </references>
+			<pkg-references>
+				<package name="nunit"/>
+			</pkg-references>
         </csc>
     </target>
     <target name="test" depends="build" if="${platform::is-win32()}">
diff -ur nant-0.85/tests/NAnt.Win32/NAnt.Win32.build nant-0.85dfsg1/tests/NAnt.Win32/NAnt.Win32.build
--- nant-0.85/tests/NAnt.Win32/NAnt.Win32.build	2006-09-03 08:59:20.000000000 +0200
+++ nant-0.85dfsg1/tests/NAnt.Win32/NAnt.Win32.build	2008-05-27 00:13:32.000000000 +0200
@@ -21,8 +21,10 @@
             <references>
                 <include name="${build.dir}/bin/NAnt.Core.dll" />
                 <include name="${build.dir}/bin/NAnt.Core.Tests.dll" />
-                <include name="${lib.framework.dir}/nunit.framework.dll" />
             </references>
+			<pkg-references>
+				<package name="nunit"/>
+			</pkg-references>
         </csc>
     </target>
     <target name="test" depends="build" if="${platform::is-win32()}">
